libyul/AsmParser.cpp

```diff
@@ -494,24 +486,9 @@ TypedName Parser::parseTypedName()
 YulString Parser::expectAsmIdentifier()
 {
 	YulString name{currentLiteral()};
-	switch (currentToken())
-	{
-	case Token::Return:
-	case Token::Byte:
-	case Token::Address:
-	case Token::Bool:
-	case Token::Identifier:
-	case Token::Var:
-	case Token::In:
-		break;
-	default:
-		expectToken(Token::Identifier);
-		break;
-	}
-
+	expectToken(Token::Identifier);
 	if (m_dialect.builtin(name))
 		fatalParserError(5568_error, "Cannot use builtin function name \"" + name.str() + "\" as identifier name.");
-	advance();
```

Comment:
The failure is due to using `expectToken` here which does advance before. Need to check if the source location is correct or if it was better before.

Reply:
```
    contract C {
      function f() public pure {
        assembly {
          function g() -> a,b, c {}
          let a, sub, mov := g()
        }
      }
    }
```

Source location after the change is the comma after `sub, `. Need to adjust the code.

Reply:
Fixed.